src/nvim/os/input.c

```diff
@@ -82,9 +82,11 @@ static void cursorhold_event(void **argv)
 
 static void create_cursorhold_event(void)
 {
-  // If the queue had any items, this function should not have been
-  // called(inbuf_poll would return kInputAvail)
-  assert(queue_empty(loop.events));
+  // If events are enabled and the queue has any items, this function should not
+  // have been called(inbuf_poll would return kInputAvail)
+  // TODO(tarruda): Cursorhold should be implemented as a timer set during the
+  // `state_check` callback for the states where it can be triggered.
+  assert(!events_enabled || queue_empty(loop.events));
```

Comment:
This assert is just to document the state of these variables. I added it because this is a confusing place to implement of cursorhold(But is required for now, cursorhold is still triggered when getting the count in `normal_execute` for example).

When [further refactoring](https://github.com/neovim/neovim/issues/3382#issuecomment-151290282) is performed, I will be able to move cursorhold triggering to the state_enter layer, possibly as a `VimState` flag that enables or disables a cursorhold timer before blocking for keys.